pkg/drivers/sqlite/sqlite.go

```diff
@@ -81,27 +63,186 @@ func NewVariant(ctx context.Context, driverName, dataSourceName string) (server.
+// changes are rolled back if an error occurs.
+func setup(dialect *generic.Generic) error {
+	ctx := context.Background()
+	txn, err := dialect.DB.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer txn.Rollback()
+
+	if err := migration(ctx, txn, dialect); err != nil {
+		return errors.Wrap(err, "migration failed")
+	}
+
+	return txn.Commit()
+}
+
+func createKineTable(txn *sql.Tx) error {
+	createTableSQL := `CREATE TABLE IF NOT EXISTS kine
+			(
+				id INTEGER PRIMARY KEY AUTOINCREMENT,
+				name TEXT NOT NULL,
+				created INTEGER,
+				deleted INTEGER,
+				create_revision INTEGER NOT NULL,
+				prev_revision INTEGER,
+				lease INTEGER,
+				value BLOB,
+				old_value BLOB
+			)`
+
+	_, err := txn.Exec(createTableSQL)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// removeTable drops the table with the given name, in the given transaction.
+func removeTable(txn *sql.Tx, tableName string) error {
+	dropTableSQL := fmt.Sprintf(`DROP TABLE IF EXISTS %s`, tableName)
+	_, err := txn.Exec(dropTableSQL)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// alterIndices drops the given old database indices and creates
+// the given new ones.
+func alterIndices(txn *sql.Tx, d *generic.Generic) error {
+	if d.LockWrites {
+		d.Lock()
+		defer d.Unlock()
+	}
+
+	dropIndices := []string{
+		`DROP INDEX IF EXISTS kine_name_index`,
+		`DROP INDEX IF EXISTS kine_name_prev_revision_uindex`,
+	}
+
+	// Drop the old indices
+	for _, stmt := range dropIndices {
+		_, err := txn.Exec(stmt)
+		if err != nil {
+			return err
+		}
+	}
+
+	createIndices := []string{
+		`CREATE INDEX IF NOT EXISTS kine_name_index ON kine (name, id)`,
+		`CREATE UNIQUE INDEX IF NOT EXISTS kine_name_prev_revision_uindex ON kine (prev_revision, name)`,
+	}
+
+	// Create the new indices
+	for _, stmt := range createIndices {
+		_, err := txn.Exec(stmt)
 		if err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
+
+// countTable counts the number of tables with the given name, in the given transaction.
+func hasTable(ctx context.Context, txn *sql.Tx, tableName string) (bool, error) {
+	tableListSQL := fmt.Sprintf(`SELECT COUNT(*) FROM sqlite_master WHERE type = 'table' AND name = '%s'`, tableName)
+	row := txn.QueryRowContext(ctx, tableListSQL)
+	var tableCount int
+	if err := row.Scan(&tableCount); err != nil {
+		return false, err
+	}
+
+	return tableCount != 0, nil
+}
+
+// migration copies rows of the old key value table to the kine
+// table ONLY IF the key value table exists and migration has not been
+// done already. It creates the kine table if it doesn't exist.
+func migration(ctx context.Context, txn *sql.Tx, d *generic.Generic) error {
+	userVersionSQL := `PRAGMA user_version`
+	row := txn.QueryRowContext(ctx, userVersionSQL)
+
+	var userVersion int
+	if err := row.Scan(&userVersion); err != nil {
+		return err
+	}
+	// No need for migration - marker has already been set
+	if userVersion == 1 {
```

Comment:
Let's make this into a `sqlite/const.go` and define it as a
```
const databaseSchemaVersion = 1
```

with appropriate documentation about what this version means. i would also suggest adding a few notes regarding guarantees we should make when bumping this version, but this is highly optional for now.